ace.d.ts

```diff
@@ -1,1170 +1,1003 @@
+            /**
+             * Removes a Range containing pos (if it exists).
+             * @param {Ace.Point} pos The position to remove, as a `{row, column}` object
+             **/
+            substractPoint(pos: Ace.Point): any;
+            /**
+             * Merges overlapping ranges ensuring consistency after changes
+             **/
+            mergeOverlappingRanges(): void;
+            rangeCount: number;
+            /**
+             * Returns a concatenation of all the ranges.
+             * @returns {Ace.Range[]}
+             **/
+            getAllRanges(): Ace.Range[];
+            /**
+             * Splits all the ranges into lines.
+             **/
+            splitIntoLines(): void;
+            /**
+             */
+            joinSelections(): void;
+            /**
+             **/
+            toggleBlockSelection(): void;
+            /**
+             *
+             * Gets list of ranges composing rectangular block on the screen
+             *
+             * @param {Ace.ScreenCoordinates} screenCursor The cursor to use
+             * @param {Ace.ScreenCoordinates} screenAnchor The anchor to use
+             * @param {Boolean} [includeEmptyLines] If true, this includes ranges inside the block which are empty due to clipping
+             * @returns {Ace.Range[]}
+             **/
+            rectangularRangeBlock(screenCursor: Ace.ScreenCoordinates, screenAnchor: Ace.ScreenCoordinates, includeEmptyLines?: boolean): Ace.Range[];
+            index?: number;
+        }
+        type AcePopupEventsCombined = Ace.EditorEvents & Ace.AcePopupEvents;
+        type AcePopupWithEditor = Ace.EventEmitter<AcePopupEventsCombined> & Ace.Editor;
+        type InlineAutocompleteAction = "prev" | "next" | "first" | "last";
+        type TooltipCommandFunction<T> = (editor: Ace.Editor) => T;
+        export interface TooltipCommand extends Ace.Command {
+            enabled: TooltipCommandFunction<boolean> | boolean;
+            getValue?: TooltipCommandFunction<any>;
+            type: "button" | "text" | "checkbox";
+            iconCssClass?: string;
+            cssClass?: string;
+        }
+        export type CommandBarTooltip = import("ace-code/src/ext/command_bar").CommandBarTooltip;
+        export type TokenizeResult = Array<Array<{
+            className?: string;
+            value: string;
+        }>>;
+    }
+    export const config: {
+        defineOptions(obj: any, path: string, options: {
+            [key: string]: any;
+        }): import("ace-code").Ace.AppConfig;
+        resetOptions(obj: any): void;
+        setDefaultValue(path: string, name: string, value: any): boolean;
+        setDefaultValues(path: string, optionHash: {
+            [key: string]: any;
+        }): void;
+        setMessages(value: any, options?: {
+            placeholders?: "dollarSigns" | "curlyBrackets";
+        }): void;
+        nls(key: string, defaultString: string, params?: {
```

Comment:
There is a conflict between this and the definition on line 72? why do we even have this duplication and the version returning any?

Reply:
Corrected differences between old config declaration in Ace namespace and generated one. (By using type alias to actual config). Copying same config type into `const config` happens on initial declaration generation, the only solution I could suggest is to manually find config node in ace.js on generator stage and replacing it with `Ace.Config` one